pulp_container/app/registry_api.py

```diff
@@ -1001,26 +985,58 @@ def put(self, request, path, pk=None):
                 tag = models.Tag.objects.get(name=tag.name, tagged_manifest=manifest)
                 tag.touch()
 
+            add_content_units += [str(tag.pk), str(manifest.pk)]
+
+            if is_manifest_list:
+                if not pending_blobs:
+                    pending_blobs = repository.pending_blobs.values_list("pk")
+
+                blobs_qs = models.Blob.objects.filter(
+                    pk__in=pending_blobs.union(found_manifests.values_list("blobs__pk"))
+                )
+                # we need to add pending blobs referenced by pending manifests
+                add_content_units += [str(blob.pk) for blob in blobs_qs]
```

Comment:
Config blobs should be included within the qs since we were adding them one by one to the `pending` list. I believe I can even remove "[str(config_blob.pk)]" from "add_content_units += [str(blob.pk) for blob in blobs_qs] + [str(config_blob.pk)]".